sklearn/base.py

```diff
@@ -747,6 +748,34 @@ def fit_predict(self, X, y=None):
         return self.fit(X).predict(X)
 
 
+class OneToOneMixin:
+    """Provides get_feature_names_out for simple transformers
+
+    Assumes there's a 1-to-1 correspondence between input features
+    and output features.
+    """
+
+    def get_feature_names_out(self, input_features=None):
+        """Get output feature names for transformation.
+
+        Returns input_features as this transformation
+        doesn't add or drop features.
+
+        Parameters
+        ----------
+        input_features : array-like of str or None, default=None
+            Input features. If None, they are generated as
+            x0, x1, ..., xn_features.
+
+        Returns
+        -------
+        feature_names : array-like of str
+            Transformed feature names.
+        """
+        return _make_feature_names(self.n_features_in_,
+                                   input_features=input_features)
```

Comment:
This is an array-like because when `input_features is not None` then this returns `input_features` without any processing.